tests/backends/test_pyevm.py

```diff
@@ -261,6 +261,37 @@ def test_from_mnemonic(self):
             assert actual.lower() == expected.lower()
             assert tester.get_balance(account=actual) == balance
 
+    def test_from_mnemonic_override_hd_path(self):
+        # Initialize PyEVM backend using MNEMONIC, num_accounts,
+        # and state overrides (balance)
+        num_accounts = 3
+        balance = to_wei(15, "ether")  # Give each account 15 Eth
+        pyevm_backend = PyEVMBackend.from_mnemonic(
+            MNEMONIC,
+            num_accounts=num_accounts,
+            genesis_state_overrides={"balance": balance},
+            hd_path="m/44'/60'/0'/7",
+        )
+
+        # Each of these accounts stems from the MNEMONIC
+        expected_accounts = [
```

Comment:
Since I don't really know what public addresses are actually expected with the provided parameters, I'm not very sure what this is testing as someone just reading the code other than that it doesn't break anything.

Maybe we could test here that the `hd_path` at least has some influence on the account generation by comparing with / without the `hd_path` kwarg. Or is there another good marker that we're testing already here? I haven't dived too deep.

Reply:
To be fair, the above note goes for the mnemonic test above this one too. It would be nice if that also checked that it at least influences the addresses generated. Maybe there's a good fix here for both if we abstract the default expected accounts into a variable at the top `DEFAULT_EXPECTED_ACCOUNTS = [...]` and add one-liners to each of these: `assert not any(acct in DEFAULT_EXPECTED_ACCOUNTS for acct in tester.get_accounts()`.

Reply:
For testing the hd_path, I've added a couple of those assertions to illustrate the accounts are different than those created with a mnemonic alone. I've left the original checks for account balances in the test above, I dont think that is covered anywhere else, but added the assertion for the default mnemonic accounts there as well.

Reply:
Oh yeah I think that works well too since `MNEMONIC` is a constant 👌🏼